src/frontend/src/components/authenticateBox.ts

```diff
@@ -77,27 +78,39 @@ export const authenticateBox = async (
   // Retry until user has successfully authenticated
   for (;;) {
     const result = await promptAuth();
-    switch (result.tag) {
-      case "ok":
-        setAnchorUsed(result.userNumber);
-        return result;
-      case "err":
-        await displayError({
-          title: result.title,
-          message: result.message,
-          detail: result.detail !== "" ? result.detail : undefined,
-          primaryButton: "Try again",
-        });
-        break;
-      case "canceled":
-        break;
-      default:
-        unreachable(result);
-        break;
+    const { success: loginData } = await handleLoginFlowResult(result);
```

Comment:
what's the case for having `success` be nullish? wouldn't something like this be more clearer?

```
{ success: LoginData } | undefined
```

Reply:
On error and when user click `Cancel`.

Reply:
Sure I understand, I'm wondering about the field; would it make sense to simplify and nullify everything, or even have `LoginData | undefined`?  or `LoginData | "canceled"` for clarity?

Reply:
I like it and find it readable as it is, otherwise I would not have coded it that way. If you want something else, please let me know what and happy to obligate. No strong opinion here.